### PR TITLE
issue: 1734068 Fix inaccurate timer_handler unregistration

### PR DESCRIPTION
**Pull request by igor-ivanov**
Special protection is needed to avoid scenario when deregistration is done
during timer_handler object destruction, timer node itself is not removed
and time for this timer node is expired. In this case there is no guarantee
to operate with timer_handler object.
So assign orig_time_msec field to maximum value;

Signed-off-by: Igor Ivanov <igor.ivanov.va@gmail.com>

### CODE REVIEW

File: src/vma/event/delta_timer.cpp [hunk 1]
```diff
@@ -234,8 +236,9 @@ void timer::process_registered_timers()
 	while (iter && (iter->delta_time_msec == 0)) {
 		tmr_logfuncall("timer expired on %p", iter->handler);
 
-		if (iter->handler){
+		if (iter->handler && !iter->lock_timer.trylock()){
```
Comment (alexvm) [inline src/vma/event/delta_timer.cpp]: Shouldn't we check that iter->handle != NULL, after we took the lock?

Comment (igor-ivanov) [reply src/vma/event/delta_timer.cpp]: I improved this.

Comment (igor-ivanov) [reply src/vma/event/delta_timer.cpp]: After thinking more my answer is "no" because `iter->handler` can not be updated after initial creation.

**Comment by igor-ivanov** [comment]
bot:retest

File: src/vma/event/delta_timer.cpp (hunk 1 above)
Comment (alexvm) [reply src/vma/event/delta_timer.cpp]: what will happen if another thread calls remove timer?

Comment (alexvm) [reply src/vma/event/delta_timer.cpp]: or this can't happen since the timer is removed only from inernal thread context, right?

Comment (igor-ivanov) [reply src/vma/event/delta_timer.cpp]: yep

**Comment by igor-ivanov** [comment]
merged as https://github.com/Mellanox/libvma/commit/e45e319b80a3be3f9650972357fc049638e52f0b

**Pull request closed without merging**